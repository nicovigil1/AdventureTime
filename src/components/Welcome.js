import React from 'react'
import Spritesheet from 'react-responsive-spritesheet';
import NextLocation from './NextLocation'

export default class Welcome extends React.Component {

  constructor() {
    super()
    this.state = {
      displayNextLocation: false,
      currentQuest: {},
      fight: false,
    }
  }
  postLocation = () => {
    navigator.geolocation.getCurrentPosition(async location => {
      let lat = location.coords.latitude
      let lon = location.coords.longitude
      let response = await fetch(`https://adventure-time-m4cap.herokuapp.com/api/v1/quest?user_id=${this.props.currentUser.ID}&lat=${lat}&long=${lon}`, {method: "POST"})
      let data = await response.json();
      this.setState({ currentQuest: data})
      this.setState({ displayNextLocation: !this.state.displayNextLocation })
    })
  }
  checkin = () => {
    navigator.geolocation.getCurrentPosition( location => {
      this.setState({currentLocation: location})
      let lat = this.state.currentLocation.coords.latitude
      let lon = this.state.currentLocation.coords.longitude
      console.log(this.props.currentUser.ID)
      // fetch(`https://adventure-time-m4cap.herokuapp.com/api/v1/checkin?user_id=${this.props.currentUser.ID}&lat=${lat}&long=${lon}`, {method: "POST"})
      fetch(`https://adventure-time-m4cap.herokuapp.com/api/v1/checkin?user_id=${this.props.currentUser.ID}&lat=39.7526897&long=-104.9962246`, {method: "POST"})
        .then(res => res.json())
        .then(res => {
          if (res["success"]) {
            console.log("success")
            this.props.setTheState({playerHealth: this.props.currentUser.current_hp})
            this.props.setTheState({fight: true})
          } else {
            alert("You're not quite there!")
          }
        })
    })
  }

  goBack = (backto) => {
    this.props.setTheState(backto)
  }

  render() {
    return(
      <div style={welcomeStyle}>
      <div style={center}>
        <Spritesheet
          id="player"
          style={stylePlayer}
          image={`https://i.ibb.co/Kjd23z4/adventurer-1-3-Sheet.png`}
          widthFrame={50}
          heightFrame={37}
          steps={60}
          fps={7}
          autoplay={true}
          loop={true} />
      </div>
        <h1>Hello {this.props.currentUser.username}!</h1>
        <div>
          <p>Your Stats</p>
          <div>
          </div>
          <ul>
            <li>Health: {this.props.currentUser.current_hp}</li>
            <li>XP: {this.props.currentUser.current_xp}</li>
          </ul>
          {!this.state.displayNextLocation && <div style={center}>
            <button onClick={this.postLocation}>Quest Info</button><br/>
            <button onClick={this.checkin}>Check In</button>
          </div>}
        </div>
        <div>
          {this.state.displayNextLocation && < NextLocation quest={this.state.currentQuest} goBack={this.goBack}/>}
        </div>
      </div>
    )
  }
}

const welcomeStyle = {
  background: "rgba(206, 215, 219, 0.96)",
  padding: "1em",
  borderRadius: "1em"
}

const stylePlayer = {
  height: '100px',
  width: '135px'
}

const center = {
  display: "flex",
  justifyContent: "center"
}