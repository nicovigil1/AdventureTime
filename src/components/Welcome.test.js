import React from 'react'
import Welcome from './Welcome'
import { shallow } from 'enzyme'
import { isMainThread } from 'worker_threads';


describe('Header', () => {
  let wrapper
  let location = ""
  let currentUser = {}
  let setTheState = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<Welcome currentUser={currentUser}
      location={location}
      setTheState={setTheState} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})