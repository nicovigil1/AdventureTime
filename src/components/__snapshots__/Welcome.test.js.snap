// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
<div
  style={
    Object {
      "background": "rgba(206, 215, 219, 0.96)",
      "borderRadius": "1em",
      "padding": "1em",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <t
      autoplay={true}
      background=""
      backgroundPosition=""
      backgroundRepeat="no-repeat"
      backgroundSize="cover"
      className=""
      direction="forward"
      endAt={false}
      fps={7}
      getInstance={[Function]}
      heightFrame={37}
      id="player"
      image="https://i.ibb.co/Kjd23z4/adventurer-1-3-Sheet.png"
      isResponsive={true}
      loop={true}
      onClick={[Function]}
      onDoubleClick={[Function]}
      onEachFrame={false}
      onEnterFrame={false}
      onInit={[Function]}
      onLoopComplete={false}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onPause={[Function]}
      onPlay={[Function]}
      onResize={false}
      startAt={0}
      steps={60}
      style={
        Object {
          "height": "100px",
          "width": "135px",
        }
      }
      timeout={0}
      widthFrame={50}
    />
  </div>
  <h1>
    Hello 
    !
  </h1>
  <div>
    <p>
      Your Stats
    </p>
    <div />
    <ul>
      <li>
        Health: 
      </li>
      <li>
        XP: 
      </li>
    </ul>
    <div
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
        }
      }
    >
      <button
        onClick={[Function]}
      >
        Quest Info
      </button>
      <br />
      <button
        onClick={[Function]}
      >
        Check In
      </button>
    </div>
  </div>
  <div />
</div>
`;
